test/invocation/chain_target.stmt

```diff
@@ -0,0 +1,161 @@
+40 columns                              |
+### Test how splits in call chain targets affect the chain.
+>>> Split function call target with unsplit chain.
+someTargetFunction(argument1, argument2).prop.method(1).method(2);
+<<<
+someTargetFunction(
+  argument1,
+  argument2,
+).prop.method(1).method(2);
+>>> Split function call target with block split chain.
+someTargetFunction(argument1, argument2)
+.prop.method(argument3, argument4, argument5);
+<<<
+someTargetFunction(
+  argument1,
+  argument2,
+).prop.method(
+  argument3,
+  argument4,
+  argument5,
+);
+>>> Split function call target with fully split chain.
+someTargetFunction(argument1, argument2, argument3)
+.method(argument).another(argument).third(argument);
+<<<
+someTargetFunction(
+      argument1,
+      argument2,
+      argument3,
+    )
+    .method(argument)
+    .another(argument)
+    .third(argument);
+>>> Split collection target with unsplit chain.
+[element1, element2, element3, element4].method().prop.another();
+<<<
+[
+  element1,
+  element2,
+  element3,
+  element4,
+].method().prop.another();
+>>> Split collection target with block split chain.
+[element1, element2, element3, element4].method().prop.another(
+argument1, argument2);
+<<<
+[element1, element2, element3, element4]
+    .method()
+    .prop
+    .another(argument1, argument2);
+>>>
+[element1, element2, element3, element4, element5].method().prop.another(
+argument1, argument2);
+<<<
+[
+  element1,
+  element2,
+  element3,
+  element4,
+  element5,
+].method().prop.another(
+  argument1,
+  argument2,
+);
```

Comment:
Is this similar to the example you had in your method chains doc?
```
// Don't split the chain, mix split and unsplit argument lists:
target.prop1.method(
  arg,
).prop2.method2(
  arg,
);
// prop2 is hidden.
```

Why do we not put `method()` and `prop` on a newline?

Reply:
The difference here is that the leading block formatted thing—in this case a list literal—is the target expression and not part of the call chain itself (which is just the `.` parts following the target). These are very similar syntactically:

```dart
foo.bar(
  argument,
).prop.another(
  argument,
);

bar(
  argument,
).prop.another(
  argument,
);
```

In the former, the target is just `foo` and `.bar()` is the first call in the chain. In the latter, the target is `bar(argument,)` and that entire call isn't considered part of the call chain. We handle those differently because in the former case, we have the option of splitting between the target and the first call:


```dart
foo
    .bar(
      argument,
    )
    .prop
    .another(
      argument,
    );
```

But in the latter, there's no way to split *before* the `bar(...)`:

```dart
???
    bar(
      argument,
    )
    .prop
    .another(
      argument,
    );
```

The best we could do is like:

```dart
bar(
      argument,
    )
    .prop
    .another(
      argument,
    );
```

Which we do sometimes, but it looks pretty weird, so the formatter tries to avoid that.

The TL;DR: is that if the target of a call chain is itself a function call (with no leading target), it *looks* like it's part of the call chain, but it's not really and we handle it kind of differently. We end up handling it much more like like a target that's a collection literal.